orion/pkg/repos/semester_repo.go

```diff
@@ -169,35 +169,44 @@ func (sr *semesterRepo) Update(semesterId string, semester domains.Semester) err
 	return handleSqlExecResult(execResult, 1, "semester was not updated")
 }
 
-func (sr *semesterRepo) Delete(semesterId string) error {
-	statement := "DELETE FROM semesters WHERE semester_id=?"
-	stmt, err := sr.db.Prepare(statement)
+func (sr *semesterRepo) Publish(semesterIds []string) error {
+	var errorString string
+
+	// Begin Transaction
+	tx, err := sr.db.Begin()
+	stmt, err := tx.Prepare("UPDATE semesters SET published_at=? WHERE semester_id=? AND published_at IS NULL")
 	if err != nil {
 		return err
 	}
 	defer stmt.Close()
 
-	execResult, err := stmt.Exec(semesterId)
-	if err != nil {
-		return err
+	now := time.Now().UTC()
+	for _, semesterId := range semesterIds {
+		_, err := stmt.Exec(now, semesterId)
+		if err != nil {
+			errorString += " " + semesterId + ": " + err.Error()
```

Comment:
We'll let @mzhang2021 merge his change in first. He has an appendError utility function that kind of does this all for you.